[WIP]Generate integration diagram

JackLeeHal opened the pull request:
Fixes # .

Changes proposed in this pull request:
- [WIP]generate component view

Commit:
[WIP]Generate integration diagram
* generate component view

jecofang01 commented on sysl2/sysl/ints.go:
```diff
@@ -0,0 +1,303 @@
+func MakeIntsDiagramVisitor(m *sysl.Module) *IntsDiagramVisitor {
+	return &IntsDiagramVisitor{
+		m:       m,
+		symbols: make(map[string]*_var),
```
DO NOT use `make`

jecofang01 commented on sysl2/sysl/ints.go:
```diff
@@ -0,0 +1,303 @@
+
+type IntsParam struct {
+	apps         []string
+	highlights   []string
```
`map[string]struct{}`

jecofang01 commented on sysl2/sysl/ints.go:
```diff
@@ -0,0 +1,303 @@
+	}
+	v.symbols[appName] = s
+	r := fmt.Sprintf("[%s] as %s", appName, alias)
+	for _, v := range highlights {
```
1. `map[string]struct{}`
2. `if _, ok := highlights[appName]; ok {
....
}`

jecofang01 commented on sysl2/sysl/ints.go:
```diff
@@ -0,0 +1,303 @@
+	var sb strings.Builder
+	if args.epa || endptAttrs["view"].GetS() == "epa" {
+		//TODO generate state view
+		fmt.Fprintln(&sb, "''''''''''''''''''''''''''''''''''''''''''")
```
duplicate with line 146 ~ line 153

jecofang01 commented on sysl2/sysl/ints.go:
```diff
@@ -0,0 +1,303 @@
+		}
+		fmt.Fprintln(&sb, "}")
+		for _, v := range params.integrations {
+			if restrictBy != "" && !stringInSlice(restrictBy, getMapKeys(mod.Apps[v.Self.Name].Attrs)) && !stringInSlice(restrictBy, getMapKeys(mod.Apps[v.Target.Name].Attrs)) {
```
`_, ok := mod.Apps[appName].Attrs[restrictBy]`

jecofang01 commented on sysl2/sysl/ints.go:
```diff
@@ -0,0 +1,303 @@
+		}
+		fmt.Fprintln(&sb, "}")
+
+		if args.clustered || endptAttrs["view"].GetS() == "clustered" {
```
Type assertion

jecofang01 commented on sysl2/sysl/ints.go:
```diff
@@ -0,0 +1,303 @@
+		fmt.Fprintln(&sb, "}")
+
+		if args.clustered || endptAttrs["view"].GetS() == "clustered" {
+			clusters := make(map[string][]string)
```
do not user `make`

jecofang01 commented on sysl2/sysl/ints.go:
```diff
@@ -0,0 +1,303 @@
+			clusters := make(map[string][]string)
+			for _, v := range params.apps {
+				cluster := strings.Split(v, " :: ")
+				if len(cluster) != 0 {
```
`len(cluster) != 0` always be true.
Refer to https://golang.org/pkg/strings/#Split
If both s and sep are empty, Split returns an empty slice.

```suggestion
      if len(cluster) > 1 {
            clusters[cluster[0]] = cluster[1:]
      }
```

jecofang01 commented on sysl2/sysl/ints.go:
```diff
@@ -0,0 +1,303 @@
+					}
+				}
+			}
+			for k, v := range clusters {
```
unneeded after above change

jecofang01 commented on sysl2/sysl/ints.go:
```diff
@@ -0,0 +1,303 @@
+					delete(clusters, k)
+				}
+			}
+			nameMap := make(map[string]string)
```
unused code

jecofang01 commented on sysl2/sysl/ints.go:
```diff
@@ -0,0 +1,303 @@
+			for k, v := range clusters {
+				fmt.Fprintf(&sb, "package \"%s\" {", k)
+				for _, n := range v {
+					appName := strings.Split(n, " :: ")[1]
```
the `n` does not contains any ` :: ` as we already split in line 175

jecofang01 commented on sysl2/sysl/ints.go:
```diff
@@ -0,0 +1,303 @@
+				appB := v.Target.Name
+				apps := []string{appA, appB}
+				var direct []string
+				for _, v := range params.highlights {
```
Refer to line 76

jecofang01 commented on sysl2/sysl/ints.go:
```diff
@@ -0,0 +1,303 @@
+				}
+				appA = strings.Split(appA, " :: ")[0]
+				appB = strings.Split(appB, " :: ")[0]
+				if appA != appB && !sliceInArray(apps, callsDrawn) && !sliceInArray(apps, callsDrawn) {
```
1. Redundant `!sliceInArray(apps, callsDrawn)`
2. _, ok := callsDrawn[apps]

jecofang01 commented on sysl2/sysl/ints.go:
```diff
@@ -0,0 +1,303 @@
+				fmt.Fprintln(&sb, "}")
+			}
+		}
+		var callsDrawn [][]string
```
callsDrawn := map[string]struct{}

jecofang01 commented on sysl2/sysl/ints.go:
```diff
@@ -0,0 +1,303 @@
+			for _, v := range params.integrations {
+				appA := v.Self.Name
+				appB := v.Target.Name
+				apps := []string{appA, appB}
```
`apps := fmt.Sprintf("%s->%s", appA, appB)`

jecofang01 commented on sysl2/sysl/ints.go:
```diff
@@ -0,0 +1,303 @@
+						}
+						fmt.Fprintf(&sb, "%s --> %s%s", VarManagerForComponent(visitor, appA, &sb, params.highlights), VarManagerForComponent(visitor, appB, &sb, params.highlights), indirect)
+						fmt.Fprintln(&sb)
+						callsDrawn = append(callsDrawn, apps)
```
callsDrawn[apps] = struct{}{}

Commit:
Update gradle bintray plugin version

Gradle gives following error when publishing maven
'message:Unable to upload files: Maven group, artifact or version defined in the pom file do not match the file path'

The issue has been fixed from 1.8.1
https://github.com/bintray/gradle-bintray-plugin/issues/242#issuecomment-397826471

Commit:
Bump version to 0.2.4

Commit:
Inline docker script

sahejsingh committed:
Add NPM auth key (#217)

* Add NPM auth key
* USE npm publish token for deploy only

Commit:
Generate integration diagram
* [WIP] generate state view

sahejsingh committed:
Fix ENEEDAUTH error: Add email to .npmrc (#218)

* Fix ENEEDAUTH error: Add email to .npmrc
* travis encrupt using -r anz-bank/sysl

Commit:
Bump version to 0.2.5

sahejsingh committed:
Esc unicode chars (#223)

* Add log level command line option
* 1) Allow \u unicode char in sysl strings, \n2) Allow reference in url params
* Add sourceContext to view
* Add test for log level

sahejsingh committed:
Release v0.2.6 (#224)

* Bump version to 0.2.6

* Remove github.com prefix from appveyor

sahejsingh committed:
Rest api param order (#225)

* Match param, query param order with legacy python

* Update package version to 1.0.4

sahejsingh committed:
Bump version to 0.2.7 (#226)

sahejsingh committed:
Add test to check gosysl windows binary (#227)

* Import paths need to have forward slash
* Add test to check gosysl windows binary
* Test gosysl binaries on travis

sahejsingh committed:
Release v0.2.8 (#228)

* Bump version to 0.2.8

* Update sysljs to 1.0.5

loki5265 committed:
Rollback urllib3 version (#230)

* Rollback version to 1.24.2

loki5265 committed:
Sysl generate sequence diagram (#209)

* Generate Sysl sequence diagrams (go)

Allenhaoli committed:
[WIP] generate state view

Allenhaoli committed:
move diagutil out of seqs

Allenhaoli committed:
Antlr version for appfmt and epfmt

Allenhaoli committed:
Add antlr grammar file for format

Allenhaoli committed:
Move diagutil out of seqs to public

Allenhaoli committed:
Refactor code of view generate
